### PR TITLE
feat(datepickers): add weekStartsOn prop on calendars

### PR DESCRIPTION
**Pull request by Axel-Duval**
## Description

Most of recent products are now supporting the "start of week" user preference. The garden calendar components set the "start of the week" according to user's locale. Currently there is no option to change this behaviour.

## Detail
By default, the start of the week is locale-based. If the `weekStartsOn` prop is specified, then it overrides the start of the week. The `weekStartsOn` prop is optional and should be a number between 0 and 6 according to the `DateFnsIndex` type.

_For instance, with `weekStartsOn = 3` the first day of week should be Wednesday_

Screenshot 2022-09-13 at 11 45 37

_For instance, with `weekStartsOn = 2` the first day of week should be Tuesday_

Screenshot 2022-09-13 at 11 46 12

## Checklist

- [ ] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [x] :globe_with_meridians: demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [ ] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [ ] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :guardsman: includes new unit tests
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: packages/datepickers/src/elements/DatepickerRange/components/Month.tsx [hunk 1]
```diff
@@ -89,14 +90,14 @@ export const Month = forwardRef<HTMLDivElement, IMonthProps>(
       [locale]
     );
 
-    const weekStartsOn = getStartOfWeek(locale);
+    const prefferedWeekStartsOn = weekStartsOn || getStartOfWeek(locale);
     const monthStartDate = startOfMonth(displayDate);
     const monthEndDate = endOfMonth(monthStartDate);
     const startDate = startOfWeek(monthStartDate, {
-      weekStartsOn
+      weekStartsOn: prefferedWeekStartsOn
```
Comment (pfiorentino-zendesk) [inline packages/datepickers/src/elements/DatepickerRange/components/Month.tsx:97]:
```suggestion
      weekStartsOn: preferredWeekStartsOn
```

File: packages/datepickers/src/elements/DatepickerRange/components/Month.tsx [hunk 2]
```diff
@@ -89,14 +90,14 @@ export const Month = forwardRef<HTMLDivElement, IMonthProps>(
...
     });
     const endDate = endOfWeek(monthEndDate, {
-      weekStartsOn
+      weekStartsOn: prefferedWeekStartsOn
```
Comment (pfiorentino-zendesk) [inline packages/datepickers/src/elements/DatepickerRange/components/Month.tsx:100]:
```suggestion
      weekStartsOn: preferredWeekStartsOn
```

**Review by jzempel — COMMENTED** [review]
@Axel-Duval nice attention to detail here. A couple of minor [nit] items:
- it would be nice if current `import` ordering were left intact
- `preferred` spelling per @pfiorentino-zendesk :pray:

**Review by jzempel — COMMENTED** [review]
lgtm. Once your build `.size-snapshot.json` gets correctly committed, we'll be good to merge and you can expect the feature to be released later this week.

**Review by Francois-Esquire — APPROVED** [review]
Thank you for the contribution @Axel-Duval  🙏

**Review by jzempel — APPROVED** [review]

**Pull request merged**